### PR TITLE
[MM-47538] Pass handler functions directly while creating API endpoints

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

https://github.com/mattermost/mattermost-server/blob/a6a44b5824e7bcf0249836b3a0b6d1d423ab24f4/web/handlers.go#L33-L40 this links API stats for insights in grafana to `func1`.

A temporary fix is to move the permission checks to inside handler functions, until a clean solution comes up to tag an API endpoint with the handler name.
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-47538
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: api4/group_local.go [hunk 1]
```diff
@@ -3,7 +3,113 @@
...
+)
+
 func (api *API) InitGroupLocal() {
-	api.BaseRoutes.Channels.Handle("/{channel_id:[A-Za-z0-9]+}/groups", api.APILocal(getGroupsByChannel)).Methods("GET")
-	api.BaseRoutes.Teams.Handle("/{team_id:[A-Za-z0-9]+}/groups", api.APILocal(getGroupsByTeam)).Methods("GET")
+	api.BaseRoutes.Channels.Handle("/{channel_id:[A-Za-z0-9]+}/groups", api.APILocal(getGroupsByChannelLocal)).Methods("GET")
+	api.BaseRoutes.Teams.Handle("/{team_id:[A-Za-z0-9]+}/groups", api.APILocal(getGroupsByTeamLocal)).Methods("GET")
+}
+
+func getGroupsByChannelLocal(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (agarciamontoro) [inline api4/group_local.go:18]: Do this function and the other one in this file contain the same code than the non-local versions except for the `requireLicense` call? If so, should we refactor them to a common helper function? 0/5

**Review by agarciamontoro — APPROVED** [review]
Looks good! I left a non-blocking comment, but this already unblocks the efforts to get the API handler name in the metrics. Thanks, @plant99!

File: api4/group_local.go (hunk 1 above)
Comment (plant99) [reply api4/group_local.go:18]: Understood, on it.

**Review by agnivade — APPROVED** [review]
Thanks @plant99 !

**Review by BenCookie95 — APPROVED** [review]

**Pull request merged**